lib/kube_link_generator.rb

```diff
@@ -82,21 +89,44 @@ def get_pods_for_role(role_name, job, options = {})
     end
   end
 
+  def patch_pod_with_imported_properties(role_name, job_name, digest)
+    client.patch_pod(
+      ENV['HOSTNAME'],
+      { metadata: { annotations: { :"skiff-in-props-#{role_name}-#{job_name}" => digest } } },
+      namespace
+    )
+  end
+
   def get_exported_properties(role_name, pod, job_name)
-    if pod.metadata.annotations["skiff-exported-properties-#{job_name}"]
+    # Exported properties are stored in a secret linked to the pod by naming convention.
+    begin
+      secret = client.get_secret("#{pod.metadata.name}-#{pod.metadata.uid}", namespace)
+    rescue
+    end
+
+    if !secret.nil?
```

Comment:
```suggestion
    unless secret.nil?
```

Reply:
Normally I would have done this, but decided to use `if !` because it is followed by a bunch of `elsif` clauses, and for some reason I find `unless ... else if ...` awkward.

Reply:
I just tried, you can't use `unless ... elsif` in Ruby, it is a syntax error:

```
SyntaxError:
  /Users/jan/suse/configgin/lib/kube_link_generator.rb:114: syntax error, unexpected elsif, expecting end
      elsif pod.metadata.annotations["sk...
      ^~~~~
```

Reply:
Ah, alright, sorry about the dumb comment then :)